config/database.yml

```diff
@@ -7,7 +7,7 @@
 default: &default
   adapter: sqlite3
   pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  timeout: 5000
+  timeout: 15000
```

Comment:
maybe 2-phase the saves, rather than trying to do the file saves inside the open connection?